expression/builtin_string.go

```diff
@@ -2970,9 +2970,76 @@ func evalNumDecArgsForFormat(f builtinFunc, row chunk.Row) (string, string, bool
 		d = formatMaxDecimals
 	}
 	dStr := strconv.FormatInt(d, 10)
+	xStr, err = roundingNum(xStr, dStr)
+	if err != nil {
+		return "", "", isNull, err
+	}
 	return xStr, dStr, false, nil
 }
 
+func roundingNum(xStr, dStr string) (string, error) {
+	if strings.Contains(xStr, ".") {
+		sign := false
+		// xStr can not has '+' prefix now.
+		// It is built in `evalNumDecArgsFormat` after evaluating `Evalxxx` method.
+		if strings.HasPrefix(xStr, "-") {
+			xStr = strings.Trim(xStr, "-")
+			sign = true
+		}
+		if strings.HasPrefix(xStr, "+") {
```

Comment:
I think we don't need this check now, the comment above is enough.

Reply:
Really? I think this will make code move strong.

Reply:
@tender-boluo Looks like this is a redundant check? Could you remove it? Because it impact the performance.